client/dashboard/task-list/tasks/connect.js

```diff
@@ -0,0 +1,140 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import apiFetch from '@wordpress/api-fetch';
+import { withDispatch } from '@wordpress/data';
+import { omit } from 'lodash';
+
+/**
+ * WooCommerce depdencies
+ */
+import { getHistory, getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal depdencies
+ */
+import { NAMESPACE } from 'wc-api/onboarding/constants';
+import withSelect from 'wc-api/with-select';
+
+class Connect extends Component {
+	componentDidMount() {
+		document.body.classList.add( 'woocommerce-admin-is-loading' );
+		const { query } = this.props;
+
+		if ( '1' === query.deny ) {
+			this.errorMessage(
+				__(
+					'You must click approve to install your extensions and connect to WooCommerce.com.',
+					'woocommerce-admin'
+				)
+			);
+			return;
+		}
+
+		if ( ! query[ 'wccom-connected' ] || ! query.request_token ) {
+			this.request();
+			return;
+		}
+		this.finish();
+	}
+
+	baseQuery() {
+		const { query } = this.props;
+		const baseQuery = omit( { ...query, page: 'wc-admin' }, [
+			'task',
+			'wccom-connected',
+			'request_token',
+			'deny',
+		] );
+		return getNewPath( {}, '/', baseQuery );
+	}
+
+	errorMessage(
+		message = __(
+			'There was an error connecting to WooCommerce.com. Please try again.',
+			'woocommerce-admin'
+		)
+	) {
+		document.body.classList.remove( 'woocommerce-admin-is-loading' );
+		getHistory().push( this.baseQuery() );
+		this.props.createNotice( 'error', message );
+	}
+
+	async request() {
+		try {
+			const connectResponse = await apiFetch( {
+				path: `${ NAMESPACE }/onboarding/plugins/request-wccom-connect`,
+				method: 'POST',
+			} );
+			if ( connectResponse && connectResponse.connectAction ) {
+				window.location = connectResponse.connectAction;
+				return;
+			}
+			throw new Error();
+		} catch ( err ) {
+			this.errorMessage();
+		}
+	}
+
+	async finish() {
+		const { query } = this.props;
+		try {
+			const connectResponse = await apiFetch( {
+				path: `${ NAMESPACE }/onboarding/plugins/finish-wccom-connect`,
+				method: 'POST',
+				data: {
+					request_token: query.request_token,
+				},
+			} );
+			if ( connectResponse && connectResponse.success ) {
+				await this.props.updateProfileItems( { wccom_connected: true } );
+				if ( ! this.props.isProfileItemsError ) {
+					// @todo Extensions are not currently installed. This will be handled on WCCOM.
+					this.props.createNotice(
+						'success',
+						__(
+							'Store connected to WooCommerce.com and extensions installed successfully.',
```

Comment:
Might be early since we don't know exactly how this will work, but I'm thinking for this message, it may be good to separate into two messages since extension installation may take longer.  Maybe something like:

```suggestion
							'Store connected to WooCommerce.com and extensions are being installed.',
```

And follow up with

`WooCommerce extensions installed successfully.`